src/nvim/if_cscope.c

```diff
@@ -1509,7 +1532,8 @@ static char *cs_parse_results(int cnumber, char *buf, int bufsize, char **contex
  */
 static void cs_file_results(FILE *f, int *nummatches_a)
 {
-  int i, j;
+  size_t i;
+  int j;
```

Comment:
In this case I think it's a good idea actually, and I can go through the other cases. However sometimes, for example when a loop counter is reused in several loops which are not nested, I would argue that declaring them at the start is more readable.